pandas/core/algorithms.py

```diff
@@ -744,39 +750,30 @@ def value_counts(
         except TypeError as err:
             raise TypeError("bins argument only works with numeric data.") from err
 
-        # count, remove nulls (from the index), and but the bins
+        # count, remove nulls (from the index), and use the bins
         result = ii.value_counts(dropna=dropna)
-        result = result[result.index.notna()]
         result.index = result.index.astype("interval")
         result = result.sort_index()
 
-        # if we are dropna and we have NO values
-        if dropna and (result._values == 0).all():
-            result = result.iloc[0:0]
-
-        # normalizing is by len of all (regardless of dropna)
-        counts = np.array([len(ii)])
-
     else:
 
         if is_extension_array_dtype(values):
 
-            # handle Categorical and sparse,
+            # handle Categorical and sparse data,
             result = Series(values)._values.value_counts(dropna=dropna)
             result.name = name
-            counts = result._values
 
         else:
             keys, counts = value_counts_arraylike(values, dropna)
 
             result = Series(counts, index=keys, name=name)
 
-    if sort:
-        result = result.sort_values(ascending=ascending)
-
     if normalize:
-        result = result / float(counts.sum())
+        counts = result._values
+        result = result / float(max(counts.sum(), 1))
```

Comment:
it looks like part of this PR is about `normalize` and another part is about `dropna`.  Could these be split into independent pieces?